### ISSUE TITLE
Pass custom marshaller to buffered client. 

### ISSUE THREAD

**Issue by makasim** [lestrrat-go/fluent-client#29]
I am evaluating the replacement of the official fluent client with your package. One thing that is missing is the ability to provide a custom marshaller for the message. I want to convert json bytes to msgpack bytes directly (without marshal\unmarshal) (https://github.com/makasim/jsontomsgp). 

I can provide a PR, just want to know if the project is still maintained and PR would be accepted.

**Comment by lestrrat** [comment]
@makasim Yeah, though it's not being actively developed but I haven't abandoned it. 

wrt to your actual feature request, I'm not entirely sure if it fits in a fluent client, but I guess I don't know enough to make a decision at this point, so PoCs/pseudocode/testcases are welcome
